src/parallel/ocbbs.cpp

```diff
@@ -1171,157 +1191,153 @@ char* BBSImpl::execute_helper(size_t* size, int id, bool exec) {
-        s = upkstr();
+    case 0: {
+        char* statement = upkstr();
         if (subworld) {
-            int size = strlen(s) + 1;
+            int size = strlen(statement) + 1;
             nrnmpi_int_broadcast(&size, 1, 0);
-            nrnmpi_char_broadcast(s, size, 0);
+            nrnmpi_char_broadcast(statement, size, 0);
         }
-        hoc_obj_run(s, nullptr);
-        delete[] s;
-        break;
+        hoc_obj_run(statement, nullptr);
+        delete[] statement;
+    } break;
     default: {
-#if 1
-        int i, j;
         size_t npickle;
-        Symbol* fname = 0;
+        Symbol* fname = nullptr;
         Object* ob = nullptr;
-        char* sarg[20];    // upto 20 argument may be strings
-        int ns = 0;        // number of args that are strings
-        int narg = 0;      // total number of args
-        if (style == 2) {  // object first
-            s = upkstr();  // template name
-            i = upkint();  // object index
-                           // printf("template |%s| index=%d\n", s, i);
-            Symbol* sym = hoc_lookup(s);
+        std::list<char*> sarg;  // Store the strings pointer to delete[] them later
+                                // Use a list because, we push pointers of the object into
+                                // the hoc stack
+        int narg = 0;           // total number of args
+        if (style == 2) {       // object first
+            char* template_name = upkstr();
+            int object_index = upkint();  // object index
+            Symbol* sym = hoc_lookup(template_name);
             if (sym) {
                 sym = hoc_which_template(sym);
             }
             if (!sym) {
-                hoc_execerror(s, "is not a template");
+                hoc_execerror(template_name, "is not a template");
             }
             hoc_Item *q, *ql;
             ql = sym->u.ctemplate->olist;
             ITERATE(q, ql) {
                 ob = OBJ(q);
-                if (ob->index == i) {
+                if (ob->index == object_index) {
                     break;
                 }
                 ob = nullptr;
             }
             if (!ob) {
-                fprintf(stderr, "%s[%d] is not an Object in this process\n", s, i);
+                fprintf(stderr,
+                        "%s[%d] is not an Object in this process\n",
+                        template_name,
+                        object_index);
                 hoc_execerror("ParallelContext execution error", 0);
             }
-            delete[] s;
-            s = upkstr();
-            fname = hoc_table_lookup(s, sym->u.ctemplate->symtable);
+            delete[] template_name;
+            char* fname_str = upkstr();
+            fname = hoc_table_lookup(fname_str, sym->u.ctemplate->symtable);
+            if (!fname) {
+                fprintf(stderr, "%s not a function in %s\n", fname_str, hoc_object_name(ob));
+                hoc_execerror("ParallelContext execution error", 0);
```

Comment:
```suggestion
                hoc_execerror("ParallelContext execution error", nullptr);
```
signature appears to be `hoc_execerror(const char* s1, const char* s2)` so I'd prefer to use a `nullptr` here explicitly.